### PR TITLE
Add the ability to change the "affects spawning" state of Humanoids

### PR DESCRIPTION
**Pull request by kashike**
Impl: https://github.com/SpongePowered/SpongeCommon/pull/297

This adds the ability to prevent humanoids from affecting spawning (activating mob spawners, and allowing mobs to spawn and despawn).

### CODE REVIEW

**Comment by gabizou** [comment]
This needs updating to master.

**Comment by kashike** [comment]
@gabizou updated to master, but it is still against `EntityPlayer` and not `EntityPlayerMP` since it is a `List<EntityPlayer>` in `World`.

File: src/main/java/org/spongepowered/api/data/key/Keys.java [hunk 1]
```diff
@@ -311,6 +311,7 @@
     public static final Key<Value<Integer>> FALL_TIME = null;
     public static final Key<Value<Boolean>> FALLING_BLOCK_CAN_HURT_ENTITIES = null;
     public static final Key<MutableBoundedValue<Double>> HEALTH_SCALE = null;
+    public static final Key<Value<Boolean>> AFFECTS_SPAWNING = null;
```
Comment (gabizou) [inline src/main/java/org/spongepowered/api/data/key/Keys.java]: Document this key please.

Comment (kashike) [inline src/main/java/org/spongepowered/api/data/key/Keys.java]: done

**Pull request closed without merging**